### ISSUE TITLE
mongodb 7.0.12 contains avx2 instructions

### ISSUE THREAD

**Issue by ondrej-zary** [GermanAizek/mongodb-without-avx#16]
mongodb upgraded mozjs to ESR115.7 in 7.0.12 and its brain-dead SConscript contains this:
`env.Append(CCFLAGS=['-mavx2'])`

Removing it does not work - looks like SIMD_avx2.cpp must be compiled with this flag. I don't know how to fix that so I hacked it around.
[mozjs-noavx.patch](https://github.com/user-attachments/files/16139083/mozjs-noavx.patch)

**Comment by zhuxb711** [comment]
@ondrej-zary  Would you share a full repo that could build a docker image that base on mongodb 7.X without AVX requirement?

**Comment by ondrej-zary** [comment]
No, I don't have any. I just build the binary and replace it in the official deb package. Then use it for unifi network (wireless controller).

**Comment by stevekerrison** [comment]
After 8 hours of building on my weak N5095 system, because I didn't think it would be _that_ bad, I also discovered this 😂.

I folded the above patch into one applied in the Dockerfile, but at the same time am experimenting with `-march=native`, because dropping AVX support shouldn't mean only supporting the generic x86-64 ISA.

Will report back in... a few more hours I guess.

**Comment by stevekerrison** [comment]
> After 8 hours of building on my weak N5095 system, because I didn't think it would be _that_ bad, I also discovered this 😂.
> 
> I folded the above patch into one applied in the Dockerfile, but at the same time am experimenting with `-march=native`, because dropping AVX support shouldn't mean only supporting the generic x86-64 ISA.
> 
> Will report back in... a few more hours I guess.

So in the end I went in the complete opposite direction and decided to just run Mongo via static QEMU under the assumption that performance really isn't a concern for my use case anyway.

If that holds true for anyone else, then I've made my simple wrapper container available: https://github.com/stevekerrison/mongo-qemu-avx

**Comment by gyrex** [comment]
> mongodb upgraded mozjs to ESR115.7 in 7.0.12 and its brain-dead SConscript contains this: `env.Append(CCFLAGS=['-mavx2'])`
> 
> Removing it does not work - looks like SIMD_avx2.cpp must be compiled with this flag. I don't know how to fix that so I hacked it around. [mozjs-noavx.patch](https://github.com/user-attachments/files/16139083/mozjs-noavx.patch)

Apologies if this is a stupid question but how do I roll this patch into the Dockerfile build?
